demo/src/task-scheduler.ts

```diff
@@ -0,0 +1,40 @@
+// Faster setTimeout(fn, 0) implementation using postMessage API
+// Based on https://dbaron.org/log/20100309-faster-timeouts
+export class MicroTaskScheduler {
+  readonly messageName = 'zero-timeout-message';
+
+  private executionQueue: Function[];
+  private stopped: boolean;
+
+  constructor() {
+    this.executionQueue = new Array<Function>();
+    this.stopped = true;
+  }
+
+  start() {
+    this.stopped = false;
+    window.addEventListener('message', this.handleMessage, true);
+  }
+
+  stop() {
+    this.stopped = true;
+    window.removeEventListener('message', this.handleMessage, true);
+  }
+
+  postTask(fn: Function) {
```

Comment:
Instead of using the generic `Function` type, it's better to be specific and write: `() => void`. Since you also use this type above, you can define it at the beginning:

export type IMicroTaskCallback = () => void;